### ISSUE TITLE
Not sure how to grab generated files and embed them as resources 

### ISSUE THREAD

**Issue by tomasherceg** [cafour/msbuild-js#1]
The package is amazing, but I am trying to grab the generated files and emit them as embedded resources in the DLL, and I am not sure how to do that.

```
  <ItemGroup>
    <PackageReference Include="Cafour.MSBuild.JS" Version="0.1.1" />
  </ItemGroup>
  ...
  <Target DependsOnTargets="JSBuild" BeforeTargets="ResolveReferences">
    <ItemGroup>
      <EmbeddedResource Include="Components\**\*.css" />
      <EmbeddedResource Include="Components\**\*.js" />
      <EmbeddedResource Include="Resources\**\*.css" />
      <EmbeddedResource Include="Resources\**\*.js" />
    </ItemGroup>
  </Target>
```

I am unable to reference the "JSBuild" target because it comes from the NuGet package. Without it, this target runs before the files are generated, and the embedded resources are missing.

**Comment by tomasherceg** [comment]
I have probably found a way, but I am not sure if I am doing it right:

```
  <PropertyGroup>
    ...
    <JSPackageManager>npm</JSPackageManager>
    <JSBuildBeforeTargets>
      GrabEmbeddedResourceFiles;
      $(JSBuildBeforeTargets)
    </JSBuildBeforeTargets>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Cafour.MSBuild.JS" Version="0.1.1" />
  </ItemGroup>

  <ItemGroup>
    <!-- Reset default files from the package -->
    <JSBuildInput Remove="$(JSWorkingDirectory)/**/*.js" />
    <JSBuildInput Remove="$(JSWorkingDirectory)/**/*.ts" />
    <JSBuildInput Remove="$(JSWorkingDirectory)/**/*.scss" />
    <JSBuildInput Remove="$(JSWorkingDirectory)/**/*.css" />
    <JSBuildInput Remove="$(JSWorkingDirectory)/**/*.json" />

    <!-- Add custom input files -->
    <JSBuildInput Include="Components/**/*.ts" />
    <JSBuildInput Include="Components/**/*.scss" />
    <JSBuildInput Include="Resources/**/*.ts" />
    <JSBuildInput Include="Resources/**/*.scss" />

    <!-- Tell Visual Studio to run the build when the included files change -->
    <UpToDateCheckInput Include="@(JSBuildInput)" />
  </ItemGroup>

  <!-- Include generated files as embedded resources -->
  <Target Name="GrabEmbeddedResourceFiles"
          BeforeTargets="PrepareForBuild">
    <ItemGroup>
      <EmbeddedResource Include="Components\**\*.css" WithCulture="false" Type="Non-Resx" />
      <EmbeddedResource Include="Components\**\*.js" WithCulture="false" Type="Non-Resx" />
      <EmbeddedResource Include="Resources\**\*.css" WithCulture="false" Type="Non-Resx" />
      <EmbeddedResource Include="Resources\**\*.js" WithCulture="false" Type="Non-Resx" />
    </ItemGroup>
  </Target>
```

I am not sure whether I am correctly replacing the default `JSBuildInput` items - maybe there is a simpler way to do that.

**Comment by cafour** [comment]
@tomasherceg Hello, I'm glad that you find this package useful. :D

**Embedded resources**

I think your solution would work but only if you ran MSBuild twice, because the `GrabEmbeddedResourceFiles` would run before `JSBuild` and thus the generated files would get noticed on the second pass.

This can be solved by using `DependsOnTargets`. See the sample I just added to README.

**Default includes**

Yup, you're removing them correctly. It's IMHO a bit of a mess, so I added the `JSUseBuildInputDefaults` and `JSUseRestoreInputDefaults` props that you can use to opt out of the default includes. It's published under 0.2.0.

**Comment by tomasherceg** [comment]
Great, it seems to be working.

I just had to add `Condition="'$(DesignTimeBuild)' != 'true'"`, otherwise it was breaking NuGet restore (as suggested in https://github.com/NuGet/Home/issues/11958):

```
  <Target Name="GrabEmbeddedResourceFiles" 
          BeforeTargets="$(JSBuildBeforeTargets)" 
          DependsOnTargets="JSBuild" 
          Condition="'$(DesignTimeBuild)' != 'true'">
  ...
  </Target>
```
